controller/konnect/konnectextension_controller.go

```diff
@@ -95,6 +98,12 @@ func (r *KonnectExtensionReconciler) SetupWithManager(ctx context.Context, mgr c
 				labelSelectorPredicate,
 			),
 		).
+		Watches(
+			&konnectv1alpha1.KonnectGatewayControlPlane{},
+			handler.EnqueueRequestsFromMapFunc(
+				enqueueKonnectExtensionsForControlPlane(mgr.GetClient()),
```

Comment:
```suggestion
				enqueueKonnectExtensionsForKonnectGatewayControlPlane(mgr.GetClient()),
```

We are going to have plain `ControlPlane` reconciliation as well soon, and being clear about the type of `ControlPlane` avoids any confusion.